google/ggrc-core
# GGRC-6389 Remove "monkey patching" for $.ajax

# Issue description

Currently in the scope of ajax_extensions.js we have patched $.ajax method. It's quite bad practice. 
Investigate possibility to avoid this solution.

# Steps to test the changes

Default handlers:
1. Launch GGRC
2. Start to create new object
3. Fill all mandatory fields
4. Turn off internet
5. Click on save btn.
6. Expected: Error internet connection was lost displays.
7. Create any object.
8. Try to create object with same title as object from previous step.
9. Expected: Error The value 'same title' is already used for another title. Title values must be unique.

# Solution description

After some investigation, I've noticed that for removing monkey patching, we need remove default handler for ajax errors and catch errors in every place where we use ajax request. This need a lot of time and solution isn't flexible. In scope of ticket I made ggrcAjax more readability and added unit test. 

# Sanity checklist




# PR Review checklist
